src/data-card-advanced.tsx

```diff
@@ -0,0 +1,498 @@
+                    <div
+                        aria-label="link"
+                        tabIndex={-1}
+                        className={sprinkles({
+                            display: 'flex',
+                        })}
+                        onClick={(event) => event.stopPropagation()}
+                        style={{
+                            marginTop: isMobile ? marginTop : marginTopButton,
+                            position: 'relative',
+                            zIndex: '2',
+                            marginLeft: -12,
+                            marginRight: -12,
+                        }}
+                    >
+                        {buttonLink}
+                    </div>
+                )}
+            </div>
+        </div>
+    );
+};
+
+type ExtraTypeof = typeof StackingGroup;
+
+type TextAs = 'h1' | 'h2' | 'h3' | 'h4';
+
+interface DataCardAdvancedProps {
+    onPress?: () => void;
+    stackingGroup?: RendersNullableElement<typeof StackingGroup>;
+    headline?: string | RendersNullableElement<typeof Tag>;
+    pretitle?: string;
+    pretitleLinesMax?: number;
+    pretitleAs?: TextAs;
+    title?: string;
+    titleLinesMax?: number;
+    titleAs?: TextAs;
+    subtitle?: string;
+    subtitleLinesMax?: number;
+    description?: string;
+    descriptionLinesMax?: number;
+    extra?: Array<RendersNullableElement<ExtraTypeof>>;
+    smallSlotSpace?: boolean;
+    button?: RendersNullableElement<typeof ButtonPrimary>;
+    footerImage?: RendersNullableElement<typeof Image>;
+    footerText?: string;
+    footerTextLinesMax?: number;
+    buttonLink?: RendersNullableElement<typeof ButtonLink>;
+    dataAttributes?: DataAttributes;
+    actions?: Array<CardAction>;
+    'aria-label'?: string;
+    onClose?: () => void;
+}
+
+export const DataCardAdvanced = React.forwardRef<HTMLDivElement, DataCardAdvancedProps>(
+    (
+        {
+            onPress,
+
+            stackingGroup,
+            headline,
+            pretitle,
+            pretitleAs,
+            pretitleLinesMax,
+            title,
+            titleAs,
+            titleLinesMax,
+            subtitle,
+            subtitleLinesMax,
+            description,
+            descriptionLinesMax,
+
+            extra,
+            smallSlotSpace,
+
+            button,
+            footerImage,
+            footerText,
+            footerTextLinesMax,
+            buttonLink,
+
+            dataAttributes,
+            actions,
+            'aria-label': ariaLabel,
+            onClose,
+        },
+        ref
+    ) => {
+        const finalActions = useTopActions(actions, onClose);
+        const hasAcations = finalActions?.length > 0;
+        const hasStackingGroup = !!stackingGroup;
+
+        const hasFooter = !!button || !!footerImage || !!footerText || !!buttonLink;
+        const footerProps = {button, footerImage, footerText, footerTextLinesMax, buttonLink};
```

Comment:
Why don't you just inline this when you call `<CardFooter` ? no need to create a new variable. The code is more difficult to read when the variables aren't declared near to where they are used, when I read `<CardFooter {...footerProps}/>` I need to scroll the whole screen to go to this object definition.

Reply:
the hasFooter const is used below the cardContentStyle, what I did was move the hasFooter above it, since the footerProps is giving an error when I try to put its content directly in the CardFooter